### DISCUSSION TITLE
Smartlog machine parsable output

### DISCUSSION THREAD

**Discussion by aCorrado** [arxanas/git-branchless#221] [Ideas]
Hi all,
i started used branchless not long ago, after being a long time lurker, and it's great.

I wonder if it would be possible to have the smartlog command output (optionally) a machine parsable representation of the graph, for example in JSON, but not necessarily.

The idea behind it is to be able to use the smartlog output in some sort of GUI

**Comment by arxanas** [comment]
Hi @aCorrado, I'm glad to hear you're finding git-branchless useful! I think adding a `--json` option would be warranted.

I don't have any plans to implement it any time soon, but if you're interested, feel free to try adding it. If you want, I can walk you through what changes would need to be made.

I'm curious if you have a GUI tool in mind?

**Reply by justhecuke** [reply]
I had a similar idea to integrate a GUI for drag-drop `git move` behavior in VS Code. Having easily parseable output would help simplify the logic to create the tree UI.

**Reply by arxanas** [reply]
Since this original answer, the [`git query` command was created](https://github.com/arxanas/git-branchless/wiki/Command:-git-query). It only prints out the commit hashes (with `--raw`, one per line), but it's usable for scripting if you use additional queries to Git to get the rest of the metadata you need.
